### ISSUE TITLE
Use `LemonTag` in `ObjectTags`

### ISSUE THREAD

**Issue by Twixes** [PostHog/posthog#9235]
We're moving away from the Ant `Tag` to our `LemonTag`. `ObjectTags` (the item tags UI) still uses Ant though. We can elevate that component by moving it to the new style.

Related: it seems we _also_ have CSS class `lemonade-tag`, which is applied to `Tag` to make it more `Lemon`. 🤔 This should be consolidated.

**Comment by thmsobrmlr** [comment]
I'm working on improving the path cleaning filters at the moment and got a bit rabbit-holed. These filters (see below) are still in ant-land and I'm positing here with the goal to surface all requirements for a lemon tag that should eventually replace them.

Path Cleaning Rules:
<img width="460" alt="existing_pills" src="https://user-images.githubusercontent.com/1851359/214023933-8878eb25-326e-4beb-9a73-008e602fae4c.png">

Object Tags:
<img width="320" alt="Screenshot 2023-01-23 at 11 43 58" src="https://user-images.githubusercontent.com/1851359/214026890-91b388af-78bc-4034-8982-e29d63b7eddf.png">

Lemon Snacks:
<img width="276" alt="Screenshot 2023-01-23 at 11 44 39" src="https://user-images.githubusercontent.com/1851359/214027179-89cd393a-09c5-490f-a64d-780d78ab5efc.png">


---

So in addition to the tags and snacks, we also have these "pills", and design wise I think that makes a lot of sense. Functionality wise there is a bit of overlap, so we might just want to have one component with different styles.

In addition to "pills" missing from Lemon UI, there is also a UX issue with the current implementation. For these tags the ordering is important, as they are applied in order backend-wise. We can't implement these tags with the sorting mechanism we have for "person property select" (see below), as they are also editable.

<img width="1051" alt="sortable_person_property_select" src="https://user-images.githubusercontent.com/1851359/214024731-1ef4b88e-0193-4c93-8f4d-10b6b2ffac9d.png">

Vice-versa, the "person property select" is sortable, but should also be editable. For example the current UX for editing filters isn't great e.g. if you are filtering by current_url and mistype, you have to remove and re-add the tag, as you can't edit it again.

In summary: It would be great to have a component for editable, sortable tags.

**Comment by Twixes** [comment]
Done by @daibhin 🙇
